### PR TITLE
DOC: GH48019 add styler to excel

### LINKED ISSUE #48019: DOC: Mention styler in the to_excel documentation
xref #47525

In our [DataFrame.to_excel](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_excel.html) documentation we don't have any mention to [Styler.to_excel](https://pandas.pydata.org/docs/reference/api/pandas.io.formats.style.Styler.to_excel.html?highlight=to_excel#pandas.io.formats.style.Styler.to_excel) which is very relevant.

Users checking the documentation on how to export a `DataFrame` to Excel should be aware that they can use the `Styler` to format the exported file. Personally I'd add the class to `See Also`, and I'd add a very short example at the end of our examples, with a link to the `Styler.to_excel` page. I think that would be good for users who may be interested in that feature. But other ideas also welcome.

### PR DESCRIPTION
**Pull request by FrnRmn**
- [x] closes #48019
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by FrnRmn** [comment]
Using scripts/validate_docstrings.py the test is passed.
> Docstring for "pandas.DataFrame.to_excel" correct. :)

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2256,6 +2256,7 @@ def to_excel(
         ExcelWriter : Class for writing DataFrame objects into excel sheets.
         read_excel : Read an Excel file into a pandas DataFrame.
         read_csv : Read a comma-separated values (csv) file into DataFrame.
+        io.formats.style.Styler.to_excel : Add styles to Excel sheet.
```
Comment (phofl) [inline pandas/core/generic.py:2259]: Can we link to Styler.to_excel instead of writing down the whole path?

Comment (FrnRmn) [reply pandas/core/generic.py:2259]: I followed what was mentioned in [#48089](https://github.com/pandas-dev/pandas/issues/48089#issuecomment-1216027713).

Comment (phofl) [reply pandas/core/generic.py:2259]: Ah got you, thx. What I meant is:

:meth:`~io.formats.style.Styler.to_excel`, which would remove the io.. path

Comment (phofl) [reply pandas/core/generic.py:2259]: You can ignore my comment, was not aware that this was done automatically

**Review by phofl — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @FrnRmn

**Pull request merged**